### DISCUSSION TITLE
v1.0.0 disclaimer and support model

### DISCUSSION THREAD

**Discussion by jonathanneo** [Snowflake-Labs/terraform-provider-snowflake#3301] [General] [answered]
Hey @sfc-gh-asawicki, 

First off, I just wanted to thank you and the team for the launch of the v1.0.0 provider!! 🚀 I've been keeping tabs on the project the past couple of months, and appreciate the huge effort the team has taken to shipping stable provider resources, whilst also resolving issues and accepting community contributions. 

My team and I are making a decision on whether we should adopt the Snowflake Terraform Provider, or if we should explore other means. As part of that decision, I wanted to clarify and understand what this disclaimer is stating: 
> ⚠️ Disclaimer: The project is in v1 version, but some features are in preview. Such resources and data sources are considered preview features in the provider, regardless of their state in Snowflake. We do not guarantee their stability. They will be reworked and marked as a stable feature in future releases. Breaking changes in these features are expected, even without bumping the major version. They are disabled by default. To use them, add the relevant feature name to preview_features_enabled field in the [provider configuration](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs#schema). The list of preview features is available below. Please always refer to the [Getting Help](https://github.com/Snowflake-Labs/terraform-provider-snowflake?tab=readme-ov-file#getting-help) section in our Github repo to best determine how to get help for your questions.

> Keep in mind that V1 does not mean we have an official Snowflake support. Please follow [creating issues guidelines](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/CREATING_ISSUES.md). 

1. "We do not guarantee their stability. Breaking changes in these features are expected, even without bumping the major version." This statement is referring specifically to "Preview" resources and doesn't apply to "Stable" resources, correct? 
2. "Keep in mind that V1 does not mean we have an official Snowflake support." What does it mean when there isn't official Snowflake support (does this mean that Snowflake may pull the project maintainers at any time to work on other things?), and will there ever be official Snowflake support?

**Comment by sfc-gh-jmichalak** [comment]
Hi @jonathanneo 👋 

Thanks a lot for your respect :) 
1. This statement refers to only preview resources. They are expected to have breaking changes in the next releases. Our plan is to rework them, provide a migration guide, and eventually mark them as stable and not in preview. Stable resources (the ones that are not listed in the `preview_features_enabled` field in the registry docs) will not have any breaking changes in v1.x.x.
2. This means that all issues are handled in this Github repository by the developers. We usually respond quickly and help with any problems, but official Snowflake Support will be available in the future with reaching General Availability. Please keep an eye on our [roadmap](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#13122024-v1-release-update).

**Reply by jonathanneo** [reply]
Thanks @sfc-gh-jmichalak for the explanation! 

Could you please elaborate more on what official Snowflake Support would/could look like with General Availability? 
- Does this mean I would be able to lodge a Support Ticket in my Snowflake Account for Snowflake TF provider support? 
- Does this mean that Snowflake would dedicate developers to maintaining the Snowflake Provider for the foreseeable future? 

I understand if you may not have the answers to some of these questions, but any information you can share would be helpful for our decision making!

**Comment by sfc-gh-dszmolka** [comment, marked as answer]
hi @jonathanneo this is David from Snowflake Support team. 'Official support' for the TF Provider means that the Support Team is ready for handling the issues incoming through the official case ingest channels, which is indeed as you mentioned : filing a Support Ticket from the Snowsight GUI support page (or, using the old method of logging into the Community portal, and doing the same from there)

Official Support Tickets have also strict SLAs and a real big bunch of other kind of procedures associated with them (to ensure timely and smoother handling for increased customer experience), which procedures simply do not exist here - they should not exist here. 

As mentioned earlier, this kind of official support is not available yet. The Support Team is not ready yet to provide this kind of official support. 
Of course, technically nothing prevents you from filing an official Support Ticket - if you do so, please expect answers which redirects you back here :) or asking your Snowflake Account Team for assistance.

As my colleague Jakub explained, the above official support (through Support Tickets) will be available with GA of the Provider. Thank you for bearing with us while we get the Support Team prepared! Until then, you already have the attention of the dev team, which is a very good thing to have :)

**Reply by jonathanneo** [reply]
Thank you for the reply @sfc-gh-dszmolka. That help clarifies a lot! 

Is it too much of a stretch to ask for some kind of indication on when the official Snowflake Support might be available or planned on your roadmap? 🙏

**Reply by sfc-gh-dszmolka** [reply]
The official Snowflake Support for the TF Provider is planned to be available at the same time when the TF Provider itself reaches General Availability status. Please give me some time to research on the planned timelines of the latter.

**Reply by sfc-gh-asawicki** [reply]
Hey @jonathanneo.

I have a little to add to what @sfc-gh-jmichalak and @sfc-gh-dszmolka have already written (thanks for the long, detailed answer, David).

One thing may be the GA projected date. As for now, we don't have a date set in stone. What I can say is that GA is our next significant milestone. We hope to be GA-ready by the end of April 2025, but multiple factors can affect this date, the main one being v1 adoption and migration problems. As we shared in [the latest roadmap entry](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/ROADMAP.md#13122024-v1-release-update), our primary goal now is to help with the migration. It will allow us to verify if there are no burning issues we must tackle before reaching GA.

We will for sure share the progress and any news as we go, so stay tuned :)

**Reply by sfc-gh-dszmolka** [reply]
If you don't have further questions now @jonathanneo on this same topic, do you think this thread can be now marked as answered? Let us know please if you still have questions. Thank you in advance !

**Comment by Visya** [comment]
Hi! Could you please clarify the "danger" of using preview features.
For example, we have had tables defined in the configuration before v1, so now they are in preview, is the only risk for us going to v1 is that for new minor versions we might have to do manual re-importing for those resources, like it has been done with deprecated resources before? Or is there more risks?

**Reply by sfc-gh-jcieslak** [reply]
Hey 👋 
Please, read the second disclaimer in [readme](https://github.com/Snowflake-Labs/terraform-provider-snowflake).
